apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RestoreObjectOptions.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+using static Google.Apis.Storage.v1.ObjectsResource;
+using static Google.Apis.Storage.v1.ObjectsResource.RestoreRequest;
+
+namespace Google.Cloud.Storage.V1;
+
+/// <summary>
+/// Options for RestoreObject operations.
+/// </summary>
+public sealed class RestoreObjectOptions
+{
+    /// <summary>
+    /// If true, the source object's access controls will be copied to the restored object,
+    /// unless the bucket has UBLA enabled, in which case the restore operation will fail.
+    /// Otherwise, the bucket's default object ACL will be used.
+    /// </summary>
+    public bool? CopySourceAcl { get; set; }
+
+    /// <summary>
+    /// The projection of the restored objct to return.
+    /// </summary>
+    public Projection? Projection { get; set; }
+
+    /// <summary>
+    /// Makes the operation conditional on whether the object's current generation
+    /// matches the given value. Setting to 0 makes the operation succeed only if
+    /// there are no live versions of the object.
+    /// </summary>
+    /// <remarks>
+    /// Note that you specify the generation value when calling the one of the restore object operations.
```

Comment:
I think we should reword this remark to clarify the difference between the value of this property and the value specified on the RestoreObject call.